docs/reference/commandline/trust_inspect.md

```diff
@@ -0,0 +1,179 @@
+
+`docker trust inspect` provides low-level JSON information on signed repositories.
+This includes all image tags that are signed, who signed them, and who can sign
+new tags.
+
+`docker trust inspect` is intended to be used for integrations into other systems, whereas `docker trust view` provides human-friendly output.
+
+`docker trust inspect` is currently experimental.
+
+
+## Examples
+
+### Get low-level details about signatures for a single image tag
+
+
+```bash
+$ docker trust inspect alpine:latest | jq
+[
+  {
+    "SignedTags": [
+      {
+        "SignedTag": "latest",
+        "Digest": "d6bfc3baf615dc9618209a8d607ba2a8103d9c8a405b3bd8741d88b4bef36478",
+        "Signers": [
+          "Repo Admin"
+        ]
+      }
+    ],
+    "AdminstrativeKeys": [
+      {
+        "Name": "Repository",
+        "Keys": [
+          "5a46c9aaa82ff150bb7305a2d17d0c521c2d784246807b2dc611f436a69041fd"
+        ]
+      },
+      {
+        "Name": "Root",
+        "Keys": [
+          "a2489bcac7a79aa67b19b96c4a3bf0c675ffdf00c6d2fabe1a5df1115e80adce"
+        ]
+      }
+    ]
+  }
+]
+```
+
+The `SignedTags` key will list the `SignedTag` name, its `Digest`, and the `Signers` responsible for the signature.
+
+`AdministrativeKeys` will list the `Repository` and `Root` keys.
+
+This format mirrors the output of `docker trust view` 
+
+If signers are set up for the repository via other `docker trust` commands, `docker trust inspect` includes a `Signers` key:
+
+```bash
+
+$ docker trust inspect my-image:purple | jq
+[
+  {
+    "SignedTags": [
+      {
+        "SignedTag": "purple",
+        "Digest": "941d3dba358621ce3c41ef67b47cf80f701ff80cdf46b5cc86587eaebfe45557",
+        "Signers": [
+          "alice",
+          "bob",
+          "carol"
+        ]
+      }
+    ],
+    "Signers": [
+      {
+        "Name": "alice",
+        "Keys": [
+          "04dd031411ed671ae1e12f47ddc8646d98f135090b01e54c3561e843084484a3",
+          "6a11e4898a4014d400332ab0e096308c844584ff70943cdd1d6628d577f45fd8"
+        ]
+      },
+      {
+        "Name": "bob",
+        "Keys": [
+          "433e245c656ae9733cdcc504bfa560f90950104442c4528c9616daa45824ccba"
+        ]
+      },
+      {
+        "Name": "carol",
+        "Keys": [
+          "d32fa8b5ca08273a2880f455fcb318da3dc80aeae1a30610815140deef8f30d9",
+          "9a8bbec6ba2af88a5fad6047d428d17e6d05dbdd03d15b4fc8a9a0e8049cd606"
+        ]
+      }
+    ],
+    "AdminstrativeKeys": [
+      {
+        "Name": "Repository",
+        "Keys": [
+          "27df2c8187e7543345c2e0bf3a1262e0bc63a72754e9a7395eac3f747ec23a44"
+        ]
+      },
+      {
+        "Name": "Root",
+        "Keys": [
+          "40b66ccc8b176be8c7d365a17f3e046d1c3494e053dd57cfeacfe2e19c4f8e8f"
+        ]
+      }
+    ]
+  }
+]
+```
+
+If the image tag is unsigned or unavailable, `docker trust inspect` does not display any signed tags.
+
+```bash
+$ docker trust inspect unsigned-img
+No signatures or cannot access unsigned-img
```

Comment:
Non-critical (and already like this for `docker trust view`, but wondering if we can return different errors for "no signatures" and "cannot access". I think that's useful information when debugging what's happening :smile:


Also, don't forget to update this output if you change the output to always return a `[]` 😅

Reply:
We intentionally left it as either because we didn't want to have any information leakage in the scenario that the user does not have access to a private repo.  We can return the `[]`, though.